### ISSUE TITLE
IIAB fails to correctly autodetect VM networking setup

### ISSUE THREAD

**Issue by m-anish** [iiab/iiab#1394]
### Expected behavior and actual behavior.

I am setting up IIAB as per the specs below. While the install proceeds smoothly, the network mode chosen is gateway, and the dhcp server sits on the public bridged interface. Ideally, it should have auto-detected the dhcp server on that interface and set it self up as an appliance. Re-running iiab-network also does not fix the problem.

### Specifications like the version of the IIAB, operating system version, or hardware details.
I am testing the IIAB one line install with this as the latest commit
b2be3ccae19bb52be80b3f399989fcafd5edca49

Setup:
Vagrant, Ubuntu 18.04 (Bionic) VM, IIAB One line install with mininal vars file
Network setup contains two interfaces
1. default NAT
2. public bridged interface

**Comment by holta** [comment]
Thanks @jvonau who's currently helping over live chat.  In reply, @m-anish has posted:
```
root@box:/opt/iiab/iiab/vars# ip r
default via 10.0.2.2 dev enp0s3 proto dhcp src 10.0.2.15 metric 100 
10.0.2.0/24 dev enp0s3 proto kernel scope link src 10.0.2.15 
10.0.2.2 dev enp0s3 proto dhcp scope link src 10.0.2.15 metric 100 
172.18.96.0/19 dev br0 proto kernel scope link src 172.18.96.1
```
And for `ip a` :
http://paste.ubuntu.com/p/qz5GrvgbK4/

And for `brctl show` :
http://paste.ubuntu.com/p/82xczbbvqw/

**Comment by holta** [comment]
Chat except:

@jvonau: I'd just remove the internal Nat from the vagrant file..  or pre-populate /etc/iiab/local_vars.yml with the change mentioned above..

@m-anish: sure. i will do that.. 
but do you think this needs to be fixed sometime?

@jvonau: Or bug ansible's devs.. 
Nothing that I'm going to automate, the overrides are there to provide the functionality wanted.

@m-anish: okay.fair enough
but this is something that can break somebodys network

**Comment by holta** [comment]
Chat excerpt:

@m-anish: do you think i could do something like this....

https://www.vagrantup.com/docs/networking/public_network.html#default-router

the "default router" section

@jvonau: What does your [vagrant] file look like?

@m-anish: https://termbin.com/6bn9

basically one extra line... config.vm.network "public_network"

the internet seems to suggest that removing the default nat interface is a) not possible or b) not a good idea

in vagrant, that is... in virtualbox one can do anything. 

so this is not a very critical problem per se.. i can just use virtualbox

@jvonau: I see that private_network is # but you get that anyway... Feels like a vagrant bug to me... As long as an interface is present it should not care about which network it is on. The user should know what the intended use of the VM is... Public or private and you have to set public...

@m-anish: well, i accept that... that it is a vagrant bug.. but in this particular case, it unearthed a iiab bug

because i dont think it is good practice to run dhcp servers on networks which already have dhcp servers.. meaning, it should not setup gateway.

it should either setup as appliance, or gracefully fail, saying that this is not a network config that i understand,so please do things manually

@jvonau: And that would be an ansible limitation..

@m-anish: (which you are telling me now anyway -- to manually set things up)

@jvonau: Ansible doesn't report the presence of two gateways..

So manual user intervention is required to override what is the detected... 

@m-anish: okay, do we need a small shell script for it?.. i will open an ansible ticket if you can tell me what to put in it

could you point me to the code in iiab, where the problem occurs in ansible?

**Comment by holta** [comment]
@jvonau clarifies:

In the [ /opt/iiab/iiab/iiab-install.log ] is the output of ansible setup a dump of the detected variables.. [0-init](https://github.com/iiab/iiab/tree/master/roles/0-init) is where the gateway gets detected check there for the variable used from ansible

**Comment by m-anish** [comment]
@jvonau says

...make a suggestion (to ansible) that under the individual network device listing that the gateway be mentioned alongside IP address and netmask and Network

**Comment by m-anish** [comment]
So, as I understand in a wider context (outside VMs as well) we do not support IIAB to work in appliance mode if it happens to be part of two independent networks, where it needs to be in appliance mode to both those networks.

**Comment by m-anish** [comment]
@jvonau  Setting iiab_lan_enabled to False does not seem to work. 
[iiab-install.log](https://github.com/iiab/iiab/files/2833209/iiab-install.log)

**Comment by holta** [comment]
> @jvonau Setting iiab_lan_enabled to False does not seem to work.
> [iiab-install.log](https://github.com/iiab/iiab/files/2833209/iiab-install.log)

Easier to read instantly in browser here: http://paste.ubuntu.com/p/C9Z4M9h8nP/

**Comment by holta** [comment]
@jvonau summary from our community/team call (http://minutes.iiab.io) today:

> “when” clauses in computed_network.yml need adjusting, “iiab-lan-enabled:” impact with Admin Console?  Did syntax change in underlying Ansible?  ipv6 popping up on some interfaces, any idea why?

@m-anish suggests:

> This is a bug, but prob not a blocker for [IIAB 6.7](https://github.com/iiab/iiab/milestone/4) release...

**Comment by holta** [comment]
@m-anish can you offer suggestions towards bringing this issue to a close — prior to IIAB 7.0's release ideally?

**Comment by holta** [comment]
@m-anish plz confirm if this is solved by PR #1906 ?  (Which is now merged.)

**Comment by holta** [comment]
@m-anish this is 10 months old and appears solved.  But if any ongoing issues remain, plz re-open and explain, Thanks!
